godot/src/Main/MainMenu.gd

```diff
@@ -9,6 +9,8 @@ const MAX_REQUEST_ATTEMPTS := 3
 # Path to the scene to load after selecting a character.
 export (String, FILE) var next_scene_path := ""
 
+signal server_request_failed
```

Comment:
Why was this change made?

Reply:
Because of this function here. I received an error of missing signal "server_request_failed" once when I failed too join world
```
# Attempts to connect to the server, then to join the world match.
func join_game_world(player_name: String, player_color: Color) -> int:
	character_menu.is_enabled = false

	var result: int = yield(ServerConnection.connect_to_server_async(), "completed")
	if result == OK:
		result = yield(ServerConnection.join_world_async(), "completed")
	if result == OK:
		# warning-ignore:return_value_discarded
		get_tree().change_scene_to(load("res://src/Main/GameWorld.tscn"))
		ServerConnection.send_spawn(player_color, player_name)
	else:
		emit_signal(
			"server_request_failed", "Error code %s: %s" % [result, ServerConnection.error_message]
		)

	character_menu.is_enabled = true
	return result
```

Reply:
This signal is never emitted, so at the moment it doesn't do anything. Also, that's not how we control the interface: the MainMenu node sets the status message on the interface directly. I can add a status box to the character panel if needed. See how the MainMenu sets the status of the LoginAndRegister interface in the rest of the code, and if you want to keep that kind of error, please use something similar for consistency.

Reply:
At first my docker server didn't properly load the lua modules as I am on win10 home and also not running it on C drive.
Thus resulting in `result = yield(ServerConnection.join_world_async(), "completed") ` != OK
which then try to emit the signal.

That's how I found out about the missing signal when godot tries to warn me as it crashes the game

Reply:
Look at your PR, you define a signal in MainMenu.gd, but it's never emitted or connected to anything, so it doesn't do anything.

And if you want to tell the player/tester there's an error, I'd recommend using the same pattern we've used in the rest of the code, that is, displaying the error message in-game, in a status bar under the relevant interface.

Reply:
The signal is emitted on line 88.
Well godot says it's missing a signal so I gave it the signal.
Probably remove it if it is not used?

Reply:
Ah I see, looks like an artifact from older versions of the project. If nothing is connected to it, it should be removed, yes.